### DISCUSSION TITLE
CET OS Support

### DISCUSSION THREAD

**Discussion by MrRinkana** [fwupd/fwupd#7578] [Q&A] [answered]
I have:
HSI-3:
`CET Platform: Supported`
Runtime suffix:
`CET OS Support: Not supported`

What does this mean and is it possible to enable CET OS support?

**Comment by superm1** [comment]
It means the hardware supports shadow stack but the OS (kernel, compiler) don't use it.

https://fwupd.github.io/libfwupdplugin/hsi.html#cet-utilized-by-os

**Reply by MrRinkana** [reply]
Is this a limitation of the Linux kernel I am using or a configurable flag (boot option etc)?

**Reply by superm1** [reply, marked as answer]
It works for me both in Fedora and Arch, you should raise a bug with your distro.

You need a new enough kernel, new enough compiler, and new enough fwupd to properly measure it.
